external/webserver/http_client.c

```diff
@@ -871,6 +871,187 @@ void http_handle_file(struct http_client_t *client, int method, const char *url,
 	}
 }
 
+static int prepare_chunk_body(char *buf, int len, const char *body, bool is_last_msg)
+{
+	int chunk_size;
+	char httpcrnl[] = "\r\n";
+	int buflen = 0;
+
+	buflen = len;
+
+	if (is_last_msg) {
+		chunk_size = 0;
+	} else {
+		if (body == NULL) {
+			HTTP_LOGE("msg body is NULL \n");
+			return -1;
+		}
+		chunk_size = strlen(body);
+	}
+
+	//append length of message
+	buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "%x", chunk_size);
+
+	//append httpcrnl
+	buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "%s", httpcrnl);
+
+	if (is_last_msg) {
+		//append httpcrnl and no body
+		buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "%s", httpcrnl);
+		return buflen;
+	}
+
+	if (HTTP_CONF_MAX_REQUEST_LENGTH - buflen - 2 <= chunk_size) {
+		HTTP_LOGE("msg body is larger than buffer can hold \n");
+		return -1;
+	}
+
+
+	//append body
+	buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "%s", body);
+
+	//append httpcrnl
+	buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "%s", httpcrnl);
+
+	return buflen;
+}
+
+static int http_send_chunk_buffer(struct http_client_t *client, char *buf)
+{
+	int  ret = 0;
+	int sndlen = 0;
+	int buflen = 0;
+
+	sndlen = strlen(buf);
+	buflen = 0;
+	while (sndlen > 0) {
+#ifdef CONFIG_NET_SECURITY_TLS
+		if (client->server->tls_init) {
+			ret = mbedtls_ssl_write(&(client->tls_ssl), (unsigned char *)buf + buflen, sndlen);
+		} else
+#endif
+		{
+			ret = send(client->client_fd, buf + buflen, sndlen, 0);
+		}
+
+		if (ret < 1) {
+			return -1;
+		} else {
+			sndlen -= ret;
+			buflen += ret;
+		}
+	}
+
+	return 0;
+}
+
+static int http_send_chunk(struct http_client_t *client, char *buf, int buflen, const char *body, bool is_last_msg)
+{
+	int ret = 0;
+
+	ret = prepare_chunk_body(buf, buflen, body, is_last_msg);
+	if (ret < 0) {
+		return -1;
+	}
+
+	ret = http_send_chunk_buffer(client, buf);
+	if (ret < 0) {
+		return -1;
```

Comment:
return ret is correct, http_send_chunk_buffer returns defined value, and it seems to be expanded in future...(HTTP_ERROR and HTTP_OK)